cloud/scope/vpc_cluster.go

```diff
@@ -686,3 +742,326 @@ func (s *VPCClusterScope) buildCOSObjectHRef() (*string, error) {
+
+	// Recommend we requeue reconciliation after subnet was successfully created
+	return true, nil
+}
+
+// buildSubnetsForZones will create a set of Subnets, using default names, for each availability zone within a Region. This is typically used when no subnets were provided, so a set of default subnets gets created.
+func (s *VPCClusterScope) buildSubnetsForZones() ([]infrav1beta2.Subnet, error) {
+	subnets := make([]infrav1beta2.Subnet, 0)
+	zones, err := s.VPCClient.GetVPCZonesByRegion(s.IBMVPCCluster.Spec.Region)
+	if err != nil {
+		return subnets, fmt.Errorf("error unknown failure retrieving zones for region %s: %w", s.IBMVPCCluster.Spec.Region, err)
+	}
+	if len(zones) == 0 {
+		return subnets, fmt.Errorf("error retrieving subnet zones, no zones found in %s", s.IBMVPCCluster.Spec.Region)
+	}
+	for _, zone := range zones {
+		name := fmt.Sprintf("%s-%s", *s.GetServiceName(infrav1beta2.ResourceTypeSubnet), zone)
+		subnets = append(subnets, infrav1beta2.Subnet{
+			Name: ptr.To(name),
+			Zone: ptr.To(zone),
+		})
+	}
+	return subnets, nil
+}
+
+// updateSubnetStatus will check the status of a IBM Cloud Subnet and update the Network Status.
+func (s *VPCClusterScope) updateSubnetStatus(subnetDetails *vpcv1.Subnet, isControlPlane bool) (bool, error) {
+	requeue := true
+	if subnetDetails.Status != nil && *subnetDetails.Status == string(vpcv1.SubnetStatusAvailableConst) {
+		requeue = false
+	}
+
+	resourceStatus := &infrav1beta2.ResourceStatus{
+		ID:   *subnetDetails.ID,
+		Name: subnetDetails.Name,
+		// Ready status will be invert of the need to requeue
+		Ready: !requeue,
+	}
+	if isControlPlane {
+		s.SetResourceStatus(infrav1beta2.ResourceTypeControlPlaneSubnet, resourceStatus)
+	} else {
+		s.SetResourceStatus(infrav1beta2.ResourceTypeWorkerSubnet, resourceStatus)
+	}
+	return requeue, nil
+}
+
+// createSubnet creates a new VPC subnet.
+func (s *VPCClusterScope) createSubnet(subnet infrav1beta2.Subnet, isControlPlane bool) error {
+	// Created resources should be placed in the cluster Resource Group (not Network, if it exists).
+	resourceGroupID, err := s.GetResourceGroupID()
+	if err != nil {
+		return fmt.Errorf("error retrieving resource group id for subnet creation: %w", err)
+	} else if resourceGroupID == "" {
+		return fmt.Errorf("error retrieving resource group id for resource group %s", s.IBMVPCCluster.Spec.ResourceGroup)
+	}
+
+	vpcID, err := s.GetVPCID()
+	if err != nil {
+		return fmt.Errorf("error retrieving vpc id for subnet creation: %w", err)
+	}
+
+	// TODO(cjschaef): Move to webhook validation.
+	if subnet.Zone == nil {
+		return fmt.Errorf("error subnet zone must be defined for subnet %s", *subnet.Name)
+	}
+
+	// NOTE(cjschaef): We likely will want to add support to use custom Address Prefixes
+	// For now, we rely on the API to assign us prefixes, as we request via IP count
+	var ipCount int64 = 256
+	// We currnetly only support IPv4
+	ipVersion := "ipv4"
```

Comment:
can this be moved to constant section in the debugging of the file?

Reply:
Sure, I moved to consts.